src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/Saml2SecurityTokenHandler.cs

```diff
@@ -0,0 +1,200 @@
+        /// element.</param>
+        /// <returns>An instance of <see cref="Saml2SecurityToken"/>.</returns>
+        /// <exception cref="InvalidOperationException">Is thrown if 'Configuration' or 'Configruation.IssuerTokenResolver' is null.</exception>
+        public override SecurityToken ReadToken(XmlReader reader)
+        {
+            XmlDocument doc = new XmlDocument();
+            XmlElement rstXml = (doc.ReadNode(reader) as XmlElement);
+            MSIdentityTokens.Saml2.Saml2SecurityToken internalSecurityToken = (MSIdentityTokens.Saml2.Saml2SecurityToken)_internalSaml2SecurityTokenHandler.ReadToken(rstXml.OuterXml);
+            TryResolveIssuerToken(internalSecurityToken.Assertion, Configuration.IssuerTokenResolver, out SecurityToken token);
+            Saml2SecurityToken saml2SecurityToken = new Saml2SecurityToken(internalSecurityToken)
+            {
+                SigningToken = token,
+                AssertionXML = rstXml.OuterXml
+            };
+            return saml2SecurityToken;
+        }
+
+        protected virtual bool TryResolveIssuerToken(Saml2Assertion assertion, SecurityTokenResolver issuerResolver, out SecurityToken token)
+        {
+            if (null == assertion)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(assertion));
+            }
+ 
+            var keyIdentifier = CoreWCF.Security.SecurityUtils.CreateSecurityKeyIdentifier(assertion.Signature.KeyInfo);
+
+            if (keyIdentifier != null
+               && issuerResolver != null)
+            {
+                if (keyIdentifier.Count < 2) //|| LocalAppContextSwitches.ProcessMultipleSecurityKeyIdentifierClauses)
+                    return issuerResolver.TryResolveToken(keyIdentifier, out token);
+                else
+                    return issuerResolver.TryResolveToken(new SecurityKeyIdentifier(keyIdentifier[0]), out token);
+            }
+            else
+            {
+                token = null;
+                return false;
+            }
+
+        }
+
+        /// <summary>
+        /// Gets a boolean indicating if the SecurityTokenHandler can Serialize Tokens. Return true by default.
+        /// </summary>
+        public override bool CanWriteToken
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Serializes the given SecurityToken to the XmlWriter.
+        /// </summary>
+        /// <param name="writer">XmlWriter into which the token is serialized.</param>
+        /// <param name="token">SecurityToken to be serialized.</param>
+        /// <exception cref="ArgumentNullException">Input parameter 'writer' or 'token' is null.</exception>
+        /// <exception cref="SecurityTokenException">The given 'token' is not a Saml2SecurityToken.</exception>
+        public override void WriteToken(XmlWriter writer, SecurityToken token)
+        {
```

Comment:
There are multiple one-liners in this file. Use expression body style methods and properties when the implementation is a trivial one-liner.